cli/src/pcluster/templates/budget_builder.py

```diff
@@ -53,35 +64,141 @@ def _add_parameters(self, budget, index):
             ),
         )
         notifications_with_subscribers = (
-            add_budget_notifications(budget) if budget.notifications_with_subscribers else None
+            self._add_budget_notifications(budget) if budget.notifications_with_subscribers else None
         )
         budget_id = f"Budget{str(index)}"
         cfn_budget = budgets.CfnBudget(
             self.scope, budget_id, budget=budget_data, notifications_with_subscribers=notifications_with_subscribers
         )
         return cfn_budget
 
+    def _add_budget_notifications(self, budget):
+        """Create the notifications with subscribers for each budget."""
+        notifications_with_subscribers = []
+        for single_notification in budget.notifications_with_subscribers:
+            subscribers = [
+                budgets.CfnBudget.SubscriberProperty(
+                    address=subscriber.address,
+                    subscription_type=subscriber.subscription_type,
+                )
+                for subscriber in single_notification.subscribers
+            ]
 
-def add_budget_notifications(budget):
-    """Create the notifications with subscribers for each budget."""
-    notifications_with_subscribers = []
-    for single_notification in budget.notifications_with_subscribers:
-        notifications_with_subscribers.append(
-            budgets.CfnBudget.NotificationWithSubscribersProperty(
-                notification=budgets.CfnBudget.NotificationProperty(
-                    comparison_operator=single_notification.notification.comparison_operator,
-                    notification_type=single_notification.notification.notification_type,
-                    threshold_type=single_notification.notification.threshold_type,
-                    threshold=single_notification.notification.threshold,
-                ),
-                subscribers=[
+            if single_notification.trigger_fleet_stop:
+                if not self.sns_arn:
```

Comment:
In which case `self.sns_arn` can be not None?

Reply:
It is defined as None at the start. It will be None until it is created, which happens if any NotificationWithSubscribers has TriggerFleetStop set to True.